sdks/python/apache_beam/typehints/native_type_compatibility.py

```diff
@@ -84,19 +84,38 @@ def _safe_issubclass(derived, parent):
     issubclass(derived, parent), or False if a TypeError was raised.
   """
   try:
-    return issubclass(derived, parent)
+    if type(derived) == typing._GenericAlias:
```

Comment:
You could consider following to reduce the nesting in try-catch.
```
  try:
    if hasattr(typing, '_GenericAlias') && (type(derived) == typing._GenericAlias):
      # Python 3.7+
      out = issubclass(derived.__origin__, parent.__origin__)
    else:
      out = issubclass(derived, parent)
    return out
  except TypeError:
    return False
```

Reply:
Do we also need to compare `issubclass(derived.__origin__, parent)`?

Also how about we add targeted unit tests just for `_safe_issublass`?

Reply:
Sounds good, will also add tests tomorrow morning.

Reply:
I have added the test. I don't think we need to add the reversed order. Partly since the test cases works great and I believe they have good coverage. Second I think the order has a purpose, we only want to match if the derived (class) is a subclass of the parent(classinfo).  See: https://docs.python.org/3/library/functions.html#issubclass. 

Let me know if you disagree and I will update :)